lib/python-sdk/common_grants_sdk/client/opportunity.py

```diff
@@ -0,0 +1,90 @@
+"""Opportunity namespace for the CommonGrants API."""
+
+import httpx
+from typing import Callable
+from uuid import UUID
+
+from .auth import Auth
+from .config import Config
+from .exceptions import raise_api_error
+from ..schemas.pydantic.models import OpportunityBase
+from ..schemas.pydantic.responses import OpportunityResponse, OpportunitiesListResponse
+
+
+class Opportunity:
+    """Class for fetching opportunity data from CommonGrants API."""
+
+    def __init__(
+        self,
+        auth: Auth,
+        config: Config,
+        http: httpx.Client,
+        url: Callable[[str], str],
+    ):
+        """Initialize the Opportunity namespace.
+
+        Args:
+            auth: Authentication instance
+            config: Configuration instance
+            http: HTTP client for making requests
+            url: Function to build full URLs from paths
+        """
+        self.auth = auth
+        self.config = config
+        self.http = http
+        self.url = url
+
+    def list(
+        self,
+        page: int,
+    ) -> OpportunitiesListResponse:
+        """Fetch a single page of opportunities.
```

Comment:
Seems like we lost the automated pagination handling. I know this was previously done with the iterator, but we don't necessarily need an iterator to handle pagination for SDK users. 

We'll want to add support for that before we release the next minor version introducing the SDK.